samples/testapp/src/androidMain/kotlin/com/android/identity/testapp/NdefService.kt

```diff
@@ -0,0 +1,217 @@
+//  configurable in IsoMdocProximitySharingScreen
+
+class NdefService: HostApduService() {
+    companion object {
+        private val TAG = "NdefService"
+
+        private var engagement: MdocNfcEngagementHelper? = null
+        private var listeningTransports = mutableListOf<MdocTransport>()
+        private var disableEngagementJob: Job? = null
+    }
+
+    private fun vibrate(pattern: List<Int>) {
+        val vibrator = ContextCompat.getSystemService<Vibrator>(
+            AndroidContexts.applicationContext,
+            Vibrator::class.java
+        )
+        vibrator?.vibrate(pattern.map { it.toLong() }.toLongArray(), -1)
+    }
+
+    private fun vibrateError() {
+        vibrate(listOf(0, 500))
+    }
+
+    private fun vibrateSuccess() {
+        vibrate(listOf(0, 100, 50, 100))
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        MainActivity.initBouncyCastle()
+    }
+
+    private var started = false
+
+    private fun startEngagement() {
+        if (disableEngagementJob != null) {
+            disableEngagementJob?.cancel()
+            disableEngagementJob = null
+        }
+        closeListeningTransports()
+
+        val eDeviceKey = Crypto.createEcPrivateKey(EcCurve.P256)
+        val timeStarted = Clock.System.now()
+
+        val presentationModel = PresentationModel.getInstance()
+        presentationModel.reset()
+        presentationModel.setWaiting()
+
+        val intent = Intent(applicationContext, NfcPresentationActivity::class.java)
+        intent.addFlags(
+            Intent.FLAG_ACTIVITY_NEW_TASK or
+                    Intent.FLAG_ACTIVITY_NO_HISTORY or
+                    Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS or
+                    Intent.FLAG_ACTIVITY_NO_ANIMATION
+        )
+        applicationContext.startActivity(intent)
+
+        val staticHandoverMethods = mutableListOf<ConnectionMethod>()
+        staticHandoverMethods.add(
+            ConnectionMethodBle(
+                supportsPeripheralServerMode = false,
+                supportsCentralClientMode = true,
+                peripheralServerModeUuid = null,
+                centralClientModeUuid = UUID.randomUUID()
+            )
+        )
+        val builder = MdocNfcEngagementHelper.Builder(
+            eDeviceKey = eDeviceKey.publicKey,
+            onHandoverComplete = { connectionMethods, encodedDeviceEngagement, handover ->
+                vibrateSuccess()
+                val duration = Clock.System.now() - timeStarted
+                listenOnMethods(connectionMethods, encodedDeviceEngagement, handover, eDeviceKey, duration)
+            },
+            onError = { error ->
+                Logger.w(TAG, "Engagement failed", error)
+                vibrateError()
+                engagement = null
+            }
+        )
+        //builder.useStaticHandover(staticHandoverMethods)
```

Comment:
Remove commented-out line? Or add a comment indicating why we're leaving it commented out?

Reply:
Done. Next PR will introduce a setting so it's possible to use both handover methods.